### PR TITLE
[C-2684 C-2955] Improve upload component hierarchy, state, and validation

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Updates component hierarchy to connect track and collection edit forms
Updates upload types and state and defines cleaner "onContinue" logic
Updates form inputs to layout more consistently
Adds collection validation
Updates track validation location and derived types

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-2684-upload-layout-validation

**Review by amendelsohn — APPROVED** [review]
Thanks for all the cleanup!!
Love the form patterns

File: packages/web/src/pages/upload-page/pages/EditPage.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    case UploadType.INDIVIDUAL_TRACK:
+    case UploadType.INDIVIDUAL_TRACKS:
+      return <EditTrackForm formState={formState} onContinue={onContinue} />
+    case UploadType.ALBUM:
+    case UploadType.PLAYLIST:
+      return (
+        <EditCollectionForm formState={formState} onContinue={onContinue} />
+      )
+  }
+  return null
```
[the remark this answers is not in the document]
Comment (dylanjeffers) [reply packages/web/src/pages/upload-page/pages/EditPage.tsx:25]: yeah good question... so it's pretty much impossible for the "uploadType" to be undefined and to be in this page, so what ill do is update the incoming form state such that it cant be the "initialState" so we wouldnt need to return null cause typescript would see all paths are covered

[the remark this answers is not in the document]
Comment (amendelsohn) [reply packages/web/src/pages/upload-page/pages/EditPage.tsx:25]: yeah that sounds great

**Pull request merged**